### PR TITLE
[api] get account module API

### PR DESCRIPTION
**Pull request by msmouse**
## Motivation

add "/state/module" api.

"/state/table_item" to follow

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?
y
## Test Plan

* unit tests
* `make` under the api folder to check against the openapi spec. Found an issue irrelevant to this though: https://github.com/aptos-labs/aptos-core/issues/515

### CODE REVIEW

**Comment by msmouse** [comment]
(Sorry, not ready for review)

**Comment by sitalkedia** [comment]
Let's change this to draft if not ready for review?

**Comment by msmouse** [comment]
@sitalkedia  Thanks.. I didn't know one can do that 😂

File: api/doc/openapi.yaml [hunk 1]
```diff
@@ -385,6 +385,74 @@ paths:
           $ref: '#/components/responses/404'
         "500":
           $ref: '#/components/responses/500'
+  /state/module/{address}/{module_name}:
```
Comment (CapCap) [inline api/doc/openapi.yaml:388]: I think ‘Account/{address}/module/{module}’ would be the restful conjugation of the existing plural ‘Account/{address}/modules’

File: api/doc/openapi.yaml [hunk 2]
```diff
@@ -385,6 +385,74 @@ paths:
...
+        latest version is used.
+      tags:
+        - state
+      parameters:
+        - $ref: '#/components/parameters/AccountAddress'
+        - name: resource_type
+          in: path
+          required: true
+          schema:
+            $ref: '#/components/schemas/MoveStructTagId'
```
[the remark this answers is not in the document]
Comment (msmouse) [reply api/doc/openapi.yaml:440]: synced offline

File: api/src/state.rs [hunk 3]
```diff
@@ -105,4 +134,19 @@ impl State {
...
         Response::new(self.latest_ledger_info, &resource)
     }
+
+    pub fn module(self, address: AccountAddress, name: Identifier) -> Result<impl Reply, Error> {
+        let module_id = ModuleId::new(address, name);
+        let access_path = AccessPath::code_access_path(module_id.clone());
+        let state_key = StateKey::AccessPath(access_path);
+        let bytes = self
+            .state_view
+            .get_state_value(&state_key)?
```
[the remark this answers is not in the document]
Comment (msmouse) [reply api/src/state.rs:144]: synced offline

File: api/doc/openapi.yaml (hunk 1 above)
Comment (msmouse) [reply api/doc/openapi.yaml:388]: good point.. moving them over

**Comment by CapCap** [comment]
Can we also update the TS SDK + tests? :-D

**Comment by msmouse** [comment]
@CapCap Briefly, how? The README.md said how it was generated, but no guidance on how to update it.

**Comment by msmouse** [comment]
@CapCap  updasted the TS sdk

**Review by CapCap — APPROVED** [review]
noice!

**Pull request merged**